### PR TITLE
Orderbook reserve - Changes following code review

### PR DESCRIPTION
**Pull request by amitkot**
@yaronvel please review discussed changes.

### CODE REVIEW

File: contracts/permissionless/Orders.sol [hunk 1]
```diff
@@ -112,44 +112,65 @@ contract Orders is Withdrawable, Utils2, OrdersInterface {
...
+                updatedSrcAmount,
+                updatedDstAmount,
+                updatedPrevId,
+                nextId)
+            ) {
                 // Let's move the order to the hinted position.
                 address maker = orders[orderId].maker;
-                removeById(orderId);
-                addAfterValidId(maker, orderId, srcAmount, dstAmount, prevId);
+                remove(orderId);
```
Comment (yaronvel) [inline contracts/permissionless/Orders.sol]: check return value

File: contracts/permissionless/Orders.sol [hunk 2]
```diff
@@ -112,44 +112,65 @@ contract Orders is Withdrawable, Utils2, OrdersInterface {
...
+                addAfterValidId(
```
Comment (yaronvel) [inline contracts/permissionless/Orders.sol]: check return value

File: contracts/permissionless/Orders.sol (hunk 1 above)
Comment (amitkot) [reply contracts/permissionless/Orders.sol]: 👍

File: contracts/permissionless/Orders.sol (hunk 2 above)
Comment (amitkot) [reply contracts/permissionless/Orders.sol]: 👍

File: contracts/permissionless/OrdersInterface.sol [hunk 3]
```diff
@@ -4,14 +4,15 @@ pragma solidity 0.4.18;
...
-    function removeById(uint32 orderId) public returns (bool);
+    function remove(uint32 orderId) public returns (bool);
     function update(uint32 orderId, uint128 srcAmount, uint128 dstAmount) public returns (bool);
     function getFirstOrder() public view returns(uint32 orderId, bool isEmpty);
     function allocateIds(uint32 howMany) public returns(uint32);
     function getTailId() public view returns(uint32);
     function getHeadId() public view returns(uint32);
     function findPrevOrderId(uint128 srcAmount, uint128 dstAmount) public view returns(uint32);
 
+
```
Comment (ilanDoron) [inline contracts/permissionless/OrdersInterface.sol]: why this extra space?
@amitkot

Comment (amitkot) [reply contracts/permissionless/OrdersInterface.sol]: Added by mistake during merge, removed them now. Thanks!

**Pull request merged**